mix_helpers.exs

```diff
@@ -704,7 +704,12 @@ defmodule Mix.Appsignal.Helper do
   end
 
   if System.otp_release() >= "23" do
-    defp tls_options, do: [versions: [:"tlsv1.3", :"tlsv1.2"]]
+    defp tls_options do
+      accepted = [:"tlsv1.3", :"tlsv1.2"]
+      supported = :ssl.versions()[:supported]
+
+      [versions: Enum.filter(accepted, &Enum.member?(supported, &1))]
```

Comment:
This could probably just be `[versions: :ssl.versions()[:supported]]` instead? That would already be `[:"tlsv1.3", :"tlsv1.2"]` on OpenSSL 1.1.1 and `[:"tlsv1.2"]` on OpenSSL 1.0.2.

I went for the "playing it safe" approach, but I'm not sure we need to play it safe here, selecting the supported ones out of a list of "accepted" TLS versions -- if a new TLS version was added tomorrow, it would only be on `:ssl.versions()[:supported]` if the cryptolib supported it, and in that case, it should be safe to list it and try to use it, even if the download mirror servers didn't support it yet. Thoughts?

Reply:
I don't have strong opinions on this. The "playing it safe" approach you implemented looks good to me, we can keep it.

Reply:
Your playing it safe would prevent any very bad and old SSL versions being used, so that's a plus, but it may also require more maintenance in the future when we need to add more versions. We could also leave it to the user to use the best versions. We don't make this decision in other integrations and just use whatever the language is compiled with. So I'm also okay with directly using `:ssl.versions()[:supported]`

Reply:
@tombruijn 

> Your playing it safe would prevent any very bad and old SSL versions being used, so that's a plus

Somewhat surprisingly, `:ssl.versions()[:supported]` already takes care of that! In this case, "supported" _also_ means "not deprecated". TLS 1.1 and their other "bad and old" friends are present in `:ssl.versions()[:available]`, but not in `:supported`.

Reply:
@unflxw Awesome! Then use that I guess :)

Reply:
Using that. 👍